hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -142,7 +143,7 @@ private void sanityCheck(Configuration conf, ResolvedSchema schema) {
 
     // validate pre_combine key
     String preCombineField = conf.get(FlinkOptions.PRECOMBINE_FIELD);
-    if (!fields.contains(preCombineField)) {
+    if (!fields.contains(getRootLevelFieldName(preCombineField))) {
       if (OptionsResolver.isDefaultHoodieRecordPayloadClazz(conf)) {
```

Comment:
We may also need to validate the nested field names.

Reply:
I agree. This change is to standardize the validations between Flink and Spark. As such, no checks on the nested field names were made.

Reply:
@danny0405 I have added the feature to validate nested fields for Flink. 

Can you please help to review this PR again? 

Thank you.